Engine/DataFeeds/ApiDataProvider.cs

```diff
@@ -40,38 +60,114 @@ public class ApiDataProvider : IDataProvider
         public ApiDataProvider()
         {
             _api = new Api.Api();
-
             _api.Initialize(_uid, _token, _dataPath);
+
+            // If we have no value for organization get account preferred
+            if (_organizationId == null)
+            {
+                var account = _api.ReadAccount();
+                _organizationId = account.OrganizationId;
+            }
+
+            // Read in data prices and organization details
+            _dataPrices = _api.ReadDataPrices(_organizationId);
+            var organization = _api.ReadOrganization(_organizationId);
+
+            // Determine if the user is subscribed to map and factor files
+            if (organization.Products.Where(x => x.Type == ProductType.Data).Any(x => x.Items.Any(x => x.Name.Contains("Factor"))))
+            {
+                _subscribedToEquityMapAndFactorFiles = true;
+            }
+
+            // Verify user has agreed to data provider agreements
+            if (organization.DataAgreement.Signed)
+            {
+                //Log Agreement Highlights
+                Log.Trace($"ApiDataProvider(): Data Terms of Use has been signed. " +
+                    $" Find agreement at: {_dataPrices.AgreementUrl}");
+                // TODO: PRINT AGREEMENT BOX
+                Thread.Sleep(TimeSpan.FromSeconds(3));
+            }
+            else
+            {
+                // Log URL to go accept terms
+                throw new Exception($"Must agree to terms at {_dataPrices.AgreementUrl}, before using the ApiDataProvider");
+            }
+
+            // Verify we have the balance to maintain our purchase limit, if not adjust it to meet our balance
+            _balance = organization.Credit.BalanceQCC;
+            if (_balance < _purchaseLimit)
+            {
+                Log.Trace("ApiDataProvider(): Purchase limit is greater than balance." +
```

Comment:
I think this should be more like a `Log.Error` or even throw argument exception?

Reply:
I'm cool with Log.Error 👍🏽  I think its good to have this behavior because if they didn't set a "Purchase Limit" , then this sets their balance as limit.